src/main/java/io/kestra/plugin/openai/AbstractTask.java

```diff
@@ -20,9 +31,34 @@ public abstract class AbstractTask extends Task implements OpenAiInterface{
 
     protected String user;
 
+    @Schema(
+        title = "The timeout value to send to the OpenAI API."
+    )
+    @PluginProperty
+    private Integer apiTimeout;
+
     protected OpenAiService client(RunContext runContext) throws IllegalVariableEvaluationException {
         String apiKey = runContext.render(this.apiKey);
 
-        return new OpenAiService(apiKey);
+        return createOpenAiService(apiKey, Duration.ofSeconds(apiTimeout));
+    }
+
+    private static OpenAiService createOpenAiService(String token, final Duration timeout) {
+        ObjectMapper mapper = OpenAiService.defaultObjectMapper();
+        mapper.addMixIn(ChatCompletion.PluginChatFunction.class, KestraChatFunctionMixin.class);
+
+        OkHttpClient client = OpenAiService.defaultClient(token, timeout);
+        Retrofit retrofit = OpenAiService.defaultRetrofit(client, mapper);
+
+        OpenAiApi api = retrofit.create(OpenAiApi.class);
+        return new OpenAiService(api);
+    }
+
+    public abstract static class KestraChatFunctionMixin {
+        @JsonIgnore
+        abstract Class<?> getParametersClass();
+
+        @JsonProperty("parameters")
+        abstract ArrayNode getParameters();
```

Comment:
This is unfortunate, maybe this issue should be brought to the OpenAI client so that a more elegant solution would be possible.
Also, can you explain in a comment why the Mixin is needed (what you describe in this comment).

Reply:
Fortunately, the OpenAI client library ended up merging a PR that made it possible to call functions without this workaround.